Added program to reverse words of string in c++.

maanpatel2492 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

https://github.com/TesseractCoding/NeoAlgo/issues/2122#issuecomment-791581760

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Pull Request has a program to display words in the reverse order of the string.It is in C++.

maanpatel2492 committed:
program to reverse string

maanpatel2492 committed:
program to reverse words of the string

maanpatel2492 committed:
Program to reverse the words

maanpatel2492 committed:
Program to reverse all the words of string

HarshCasper commented:
Add the Issue Link in your PR Description. The PR would be reviewed after the Coding Period of GSSoC starts. Thanks!

Simi06 commented:
Can I please have this assigned to me?

Vishakha-sta commented on Rev_str.cpp:
```diff
@@ -0,0 +1,37 @@
+    getline(cin, str);
+    rev_words(str);
+    return 0;
+}
```
/*
test case 1:
input:- 

Hello World!

output:-

World! Hello

test case 2:
input:- 

Hello     .        World!

output:-

World!        .     Hello

*/

HarshCasper reviewed (changes requested):
Few changes: 

- Add a beginner friendly pitch at the top of the Code.
- Add Time and Space Complexity at the end of the Code
- Add sample I/O at the end of the Code
- Add comments where possible

maanpatel2492 commented:
> Few changes:
> 
> * Add a beginner friendly pitch at the top of the Code.
> * Add Time and Space Complexity at the end of the Code
> * Add sample I/O at the end of the Code
> * Add comments where possible

Sure.

atarax665 commented:
@maanpatel2492  Are you still working on this?

maanpatel2492 commented:
> @maanpatel2492 Are you still working on this?

Yes,
I have to make above changes and create pull request again right?

HarshCasper commented:
> > @maanpatel2492 Are you still working on this?
> 
> Yes,
> I have to make above changes and create pull request again right?

Make the changes in the same PR

maanpatel2492 committed:
Updates Rev_str.cpp

Added beginner friendly pitch , Time and space complexity , comments and test case .

ankitaggarwal23 commented:
@maanpatel2492 All contributions done by participants from 8 March in GSSoC'21 Projects will be counted. Thanks, Closing this as of now.

maanpatel2492 commented:
> @maanpatel2492 All contributions done by participants from 8 March in GSSoC'21 Projects will be counted. Thanks, Closing this as of now.


There is invalid tag.Does I made any mistake.

ankitaggarwal23 commented:
Code plagiarized from [here](https://www.geeksforgeeks.org/reverse-words-in-a-given-string/)
